junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/AtypicalJvmMethodNameTests.java

```diff
@@ -31,15 +32,18 @@
 
 	@Test
 	void kotlinTestWithMethodNameContainingSpecialCharacters() {
-		ExecutionEventRecorder eventRecorder = executeTestsForClass(ArbitraryNamingKotlinTestCase.class);
-		assertThat(eventRecorder.getTestFinishedCount()).isEqualTo(2);
+		ExecutionsResult executionsResult = executeTestsForClass(
+			ArbitraryNamingKotlinTestCase.class).getExecutionsResult();
+		assertThat(executionsResult.getTestsStartedCount()).isEqualTo(2);
 
-		TestDescriptor testDescriptor1 = eventRecorder.getSuccessfulTestFinishedEvents().get(0).getTestDescriptor();
+		TestDescriptor testDescriptor1 = executionsResult.getEventsFinished(TestExecutionResult.Status.SUCCESSFUL).get(
```

Comment:
From `.getSuccessfulTestFinishedEvents()` to `.getEventsFinished(TestExecutionResult.Status.SUCCESSFUL)`.

Please provide the old method for convenience and for less noise in the PR diff via a simple forwarding impl:

```
ExecutionEventRecorder::getSuccessfulTestFinishedEvents() {
  return getEventsFinished(TestExecutionResult.Status.SUCCESSFUL);
}
```

This applies to _all_ code changes found in this PR as well.

Reply:
Sure thing!

So instead of having this diff be breaking, I can add them back, but should I either provide an `@Deprecated` or would a simple javadoc referencing logic similarity be enough?

Reply:
> [...] simple javadoc referencing logic similarity [...]

This should suffice.